src/rnpkeys/rnpkeys.cpp

```diff
@@ -254,6 +254,10 @@ rnp_cmd(rnp_cfg_t *cfg, cli_rnp_t *rnp, optdefs_t cmd, const char *f)
     case CMD_GENERATE_KEY: {
         if (f == NULL) {
             f = rnp_cfg_getstr(cfg, CFG_USERID);
+            if(rnp_cfg_getstr(cfg, CFG_USERID, 2) != NULL) {
```

Comment:
You should use `rnp_cfg_getlist()` here, to quickly check it's length. Also here or later you may use more C++-ish wrapper `rnp_cfg_copylist_string()`.